lms/templates/learner_dashboard/course_enroll_2017.underscore

```diff
@@ -1,4 +1,14 @@
-<% if (!is_enrolled) { %>
+<% if (is_enrolled && (typeof expired === 'undefined' || expired === false)) { %>
+    <% if (is_enrollment_open || is_course_ended) { %>
```

Comment:
This still isn't quite right. If these are both false, we still want to show View Course. You can simplify here. If the user is enrolled (unexpired), show View Course in all cases unless the course is ended.